src/usr/local/www/services_dhcp.php

```diff
@@ -235,6 +241,56 @@ function validate_partial_mac_list($maclist) {
 
 	/* input validation */
 
+	/*
+	 * Check the OMAPI settings
+	 * - Make sure that if the port is defined, that it is valid and isn't in use
+	 * - Make sure the key length is at least 256bit (45 characters)
+	 * - Generate a new key if selected
+	 */
+	if($_POST['omapi_port'] && $_POST['omapi_port'] != "") {
+		// Check the port entry
+		if(in_array($_POST['omapi_port'], range(1, 65535))) {
```

Comment:
We have an `is_port($port)` function in `util.inc` that does a better check to tell if a submitted value is a port number, you should use it instead.